### ISSUE TITLE
flake8-pydantic crashes when `myvar: mytype =` encountered

### ISSUE THREAD

**Issue by crivetimihai** [Viicos/flake8-pydantic#8]
Issues with flake8==7.0.0 and flake8-pydantic==0.3.0 on Python 3.11.2 Linux - flake8 crashes when certain code is encountered, such as: `module_args: dict = dict`. Other similar declarations crash it as well, for example: `my_var: str = "test"`

## Example code:
```
def run_module() -> None:
    module_args: dict = dict(
        prompt=dict(type="str", required=True),
    )

    result = dict(changed=False, response="")
    pass
```

## Example Trace:

```
Traceback (most recent call last):
  File "~/venv/bin/flake8", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "~/venv/lib/python3.11/site-packages/flake8/main/cli.py", line 23, in main
    app.run(argv)
  File "~/venv/lib/python3.11/site-packages/flake8/main/application.py", line 198, in run
    self._run(argv)
  File "~/venv/lib/python3.11/site-packages/flake8/main/application.py", line 187, in _run
    self.run_checks()
  File "~/venv/lib/python3.11/site-packages/flake8/main/application.py", line 103, in run_checks
    self.file_checker_manager.run()
  File "~/venv/lib/python3.11/site-packages/flake8/checker.py", line 238, in run
    self.run_serial()
  File "~/venv/lib/python3.11/site-packages/flake8/checker.py", line 216, in run_serial
    self.results = [
                   ^
  File "~/venv/lib/python3.11/site-packages/flake8/checker.py", line 221, in <listcomp>
    ).run_checks()
      ^^^^^^^^^^^^
  File "~/venv/lib/python3.11/site-packages/flake8/checker.py", line 526, in run_checks
    self.run_ast_checks()
  File "~/venv/lib/python3.11/site-packages/flake8/checker.py", line 428, in run_ast_checks
    for line_number, offset, text, _ in runner:
  File "~/venv/lib/python3.11/site-packages/flake8_pydantic/plugin.py", line 20, in run
    visitor.visit(self._tree)
  File "/usr/lib/python3.11/ast.py", line 418, in visit
    return visitor(node)
           ^^^^^^^^^^^^^
  File "/usr/lib/python3.11/ast.py", line 426, in generic_visit
    self.visit(item)
  File "/usr/lib/python3.11/ast.py", line 418, in visit
    return visitor(node)
           ^^^^^^^^^^^^^
  File "/usr/lib/python3.11/ast.py", line 426, in generic_visit
    self.visit(item)
  File "/usr/lib/python3.11/ast.py", line 418, in visit
    return visitor(node)
           ^^^^^^^^^^^^^
  File "~/venv/lib/python3.11/site-packages/flake8_pydantic/visitor.py", line 135, in visit_AnnAssign
    self._check_pyd_001(node)
  File "~/venv/lib/python3.11/site-packages/flake8_pydantic/visitor.py", line 36, in _check_pyd_001
    self.current_class in {"pydantic_model", "dataclass"}
    ^^^^^^^^^^^^^^^^^^
  File "~/venv/lib/python3.11/site-packages/flake8_pydantic/visitor.py", line 32, in current_class
    return self.class_stack[-1]
           ~~~~~~~~~~~~~~~~^^^^
IndexError: deque index out of range

```

## Mitigation

Making a change to `visitor.py` does help, as in, it doesn't crash anymore:
```
    def current_class(self) -> ClassType:
        try:
            return self.class_stack[-1]
        except IndexError:
		        return
```

Though I believe this is not sufficient, it probably skips that element.

**Comment by Viicos** [comment]
Thanks for the report, I'll take a look and come with a fix as soon as I can. I'll probably try running the plugin on large code bases to avoid such crashes in the future.

**Comment by Viicos** [comment]
@crivetimihai I published `0.3.1` which fixes the issue.

**Comment by crivetimihai** [comment]
Tested, works!
